src/lib/child_processes/termination.ml

```diff
@@ -9,20 +9,12 @@ include Hashable.Make_binable (Pid)
 type process_kind = Prover | Verifier | Libp2p_helper
 [@@deriving show { with_path = false }, yojson]
 
-type data = { kind : process_kind; termination_expected : bool }
-[@@deriving yojson]
-
-type t = data Pid.Table.t
+type t = process_kind Pid.Table.t
 
 let create_pid_table () : t = Pid.Table.create ()
 
-let register_process ?(termination_expected = false) (t : t) process kind =
-  let data = { kind; termination_expected } in
-  Pid.Table.add_exn t ~key:(Process.pid process) ~data
-
-let mark_termination_as_expected t child_pid =
-  Pid.Table.change t child_pid
-    ~f:(Option.map ~f:(fun r -> { r with termination_expected = true }))
+let register_process (t : t) process kind =
```

Comment:
Does it make sense to call take a process here instead, and `wait_safe` for it, then call `remove` when that wait triggers?

Reply:
There should be just one `wait_safe` per process, right? 

We want to do different things for different processes when they terminate. For libp2p and the verifier, we want to remove the pid from the table, and also close stdin. For the prover, we want to crash the daemon. So calling the `wait_safe` when registering the process doesn't accommodate this variation.

Reply:
I suppose you could pass a thunk or lazy to the function to invoke/force when the `wait_safe` returns.

Reply:
Multiple `wait_safe`'s shouldn't be a problem: any subsequent one will just resolve to the `Deferred.t` representing the first syscall. Any associated errors will be caught by different monitors for the different calls on the same process, but I can't see that being an issue.

I think this PR is fine as-is though, no need to make the change now.